ert_shared/ensemble_evaluator/entity/prefect_ensamble.py

```diff
@@ -0,0 +1,287 @@
+                    "stage_id": self.get_id(iens, stage_name=stage["name"]),
+                    "step_id": self.get_id(iens, stage_name=stage["name"], step_name=step["name"]),
+                    **step})
+
+        produced = set()
+        ordering = []
+        while table_of_elements:
+            temp_list = produced.copy()
+            for element in table_of_elements:
+
+                if set(element.get("inputs", [])).issubset(temp_list):
+                    ordering.append(element)
+                    produced = produced.union(set(element["outputs"]))
+                    table_of_elements.remove(element)
+        return ordering
+
+    def get_id(self, iens, stage_name, step_name=None, job_index=None):
+        real = next(x for x in self._reals if x.get_iens() == iens)
+        stage = next(x for x in real.get_stages() if x.get_name() == stage_name)
+        if step_name is None:
+            return stage.get_id()
+        step = next(x for x in stage.get_steps() if x.get_name() == step_name)
+        if job_index is None:
+            return step.get_id()
+        job = step.get_jobs()[job_index]
+        return job.get_id()
+
+    def run_flow(self, dispatch_url, coef_input_files, real_range):
+        with Flow('Test jobs') as flow:
+            for iens in real_range:
+                o_t_res = {}
+                for step in self.get_ordering(iens=iens):
+                    job_list = [{
+                        "id": self.get_id(iens, step["stage_name"], step["name"], idx),
+                        "name": job["name"],
+                        "executable": job["executable"],
+                        "args": list(map(str, job["args"])),
+                    } for idx, job in enumerate(step["jobs"])]
+                    inputs = [o_t_res.get(i, []) for i in step.get("inputs", [])]
+
+                    stage_task = RunProcess(
+                        resources=step.get("resources", []) + [coef_input_files[iens]],
+                        outputs=step.get("outputs", []),
+                        job_list=job_list,
+                        iens=iens,
+                        cmd="python",
+                        url=dispatch_url,
+                        step_id=step["step_id"],
+                        stage_id=step["stage_id"],
+                        on_failure=partial(self._on_task_failure, url=dispatch_url)
+                    )
+                    result = stage_task(expected_res=inputs)
+
+                    for o in step.get("outputs", []):
+                        o_t_res[o] = result["outputs"]
+        return flow.run()
+        # return flow.run(executor=DaskExecutor(address="tcp://192.168.100.6:8786"))
+
+    def evaluate(self, host, port):
+        # executor = DaskExecutor(
+        #     cluster_class="dask_jobqueue.LSFCluster",
+        #     cluster_kwargs=cluster_kwargs,
+        #     debug=True,
+        # )
+        dispatch_url = f"ws://{host}:{port}/dispatch"
+        wait_for_ws(dispatch_url)
```

Comment:
I wonder if we could make the `EnsembleEvaluator`guarantee that the ws would be up and running at this point?

Reply:
That sounds reasonable